src/services/LbpProjectTokenPriceService.ts

```diff
@@ -0,0 +1,144 @@
+import { autoinject } from "aurelia-framework";
+import * as moment from "moment-timezone";
+import Moment = moment.Moment;
+
+@autoinject
+export class LbpProjectTokenPriceService {
+
+  maxSupply: number;
+  projectTokenInitialLbpAmount: number;
+  fundingTokenInitialLbpAmount: number;
+  startWeightProjectToken: number;
+  endWeightProjectToken: number;
+  startDate: string;
+  endDate: string;
+  roundedStartDate: Moment;
+  roundedEndDate: Moment;
+  lbpDurationInHours: number;
+  pricePerFundingToken: number;
+
+
+  constructor(
+    maxSupply: number,
+    projectTokenInitialLbpAmount: number,
+    fundingTokenInitialLbpAmount: number,
+    startWeightProjectToken: number,
+    endWeightProjectToken: number,
+    startDate: string,
+    endDate: string,
+    pricePerFundingToken: number,
+  ) {
+    this.maxSupply = maxSupply;
+    this.projectTokenInitialLbpAmount = projectTokenInitialLbpAmount;
+    this.fundingTokenInitialLbpAmount = fundingTokenInitialLbpAmount;
+    this.startWeightProjectToken = startWeightProjectToken;
+    this.endWeightProjectToken = endWeightProjectToken;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.roundedStartDate = moment(this.startDate).startOf("hour");
+    this.roundedEndDate = moment(this.endDate).startOf("hour");
+    this.lbpDurationInHours = moment(this.roundedEndDate).diff(this.roundedStartDate, "hours");
+    this.pricePerFundingToken = pricePerFundingToken;
+  }
+
+  getMarketCap(
+    projectTokenInPool: number,
+    fundingTokenInPool: number,
+    projectTokenWeight: number,
+  ): number {
+    const projectTokenMcap =
+      this.getPriceAtWeight(
+        projectTokenInPool,
+        fundingTokenInPool,
+        projectTokenWeight,
+        this.pricePerFundingToken,
+      ) *
+      this.maxSupply;
+
+
+    return Math.round(projectTokenMcap);
+  }
+
+  getPriceAtWeight(
+    projectTokenInPool: number,
+    fundingTokenInPool: number,
+    projectTokenWeight: number,
+    pricePerFundingToken: number,
+  ): number {
+    const fundingTokenValue = fundingTokenInPool * pricePerFundingToken;
```

Comment:
I think you should be using BIgNumber methods.  You will need to use our BigNumberService (BigNumber.js) because the usual ethers BigNumber doesn't do division.

You may need to account for the # of decimals of each token.  You can talk to the contract guys about that.